### PR TITLE
Add VoQ Recirc interface (i.e., Ethernet-Rec) to interface maps for S…

### PR DESCRIPTION
**Pull request by ysmanman**
…yncD-connected MIB(s)

**- What I did**

VoQ inband port support was added in https://github.com/Azure/sonic-snmpagent/pull/228. This PR is to add support for VoQ recirc port.

This PR depends https://github.com/Azure/sonic-py-swsssdk/pull/118, which defines index for VoQ recirc port.

**- How I did it**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: src/sonic_ax_impl/mibs/__init__.py [hunk 1]
```diff
@@ -284,7 +285,8 @@ def init_sync_d_interface_tables(db_conn):
     for sai_id, if_name in if_id_map_util.items():
         if (re.match(port_util.SONIC_ETHERNET_RE_PATTERN, if_name) or \
                 re.match(port_util.SONIC_ETHERNET_BP_RE_PATTERN, if_name) or \
-                re.match(port_util.SONIC_ETHERNET_IB_RE_PATTERN, if_name)):
+                re.match(port_util.SONIC_ETHERNET_IB_RE_PATTERN, if_name) or \
+                re.match(port_util.SONIC_ETHERNET_REC_RE_PATTERN, if_name)):
```
Comment (SuvarnaMeenakshi) [inline src/sonic_ax_impl/mibs/__init__.py:289]: Can we add unit-test for both the new interface patterns?

**Comment by SuvarnaMeenakshi** [comment]
Can you raise PR to update sub-module to include : https://github.com/Azure/sonic-py-swsssdk/pull/118 so that snmpagent build succeeds.

**Comment by ysmanman** [comment]
> Can you raise PR to update sub-module to include : [Azure/sonic-py-swsssdk#118](https://github.com/Azure/sonic-py-swsssdk/pull/118) so that snmpagent build succeeds.

@SuvarnaMeenakshi Thanks for the reminder. Created PR https://github.com/Azure/sonic-buildimage/pull/10220

File: src/sonic_ax_impl/mibs/__init__.py (hunk 1 above)
Comment (ysmanman) [reply src/sonic_ax_impl/mibs/__init__.py:289]: @SuvarnaMeenakshi sorry, I missed this comment.  Wondering do you know if there is an existing unit test that I can start with?

Comment (SuvarnaMeenakshi) [reply src/sonic_ax_impl/mibs/__init__.py:289]: @ysmanman You could add the IB in asic mock db, like the mock data added here:
https://github.com/Azure/sonic-utilities/blob/master/tests/mock_tables/asic0/appl_db.json#L50

Comment (ysmanman) [reply src/sonic_ax_impl/mibs/__init__.py:289]: @SuvarnaMeenakshi Thanks for the pointer!

File: tests/namespace/test_mibs.py [hunk 2]
```diff
@@ -36,6 +36,17 @@ def test_init_namespace_sync_d_lag_tables(self):
         self.assertTrue("PortChannel_Temp" in lag_name_if_name_map)
         self.assertTrue(lag_name_if_name_map["PortChannel_Temp"] == [])
 
+    def test_init_sync_d_interface_tables_for_recirc_ports(self):
```
Comment (SuvarnaMeenakshi) [inline tests/namespace/test_mibs.py:39]: thank you, will be good to add test case to verify the oid output like in test_interfaces/test_pfc/test_queues

Comment (SuvarnaMeenakshi) [inline tests/namespace/test_mibs.py:39]: thank you, will be good to add test case to verify the oid output like in test_interfaces/test_pfc/test_queues

Comment (SuvarnaMeenakshi) [reply tests/namespace/test_mibs.py:39]: For ex:
Interfaces MIB with this change:
iso.3.6.1.2.1.2.2.1.2.177 = STRING: "Ethernet6/45/1"
..
iso.3.6.1.2.1.2.2.1.2.10000 = STRING: "Management1/1"
**iso.3.6.1.2.1.2.2.1.2.11000 = STRING: "Recirc6/0/1"
iso.3.6.1.2.1.2.2.1.2.12000 = STRING: "Recirc6/0/0"**

**Comment by arlakshm** [comment]
@ysmanman The test `test_init_sync_d_interface_tables_for_recirc_ports` is failing. Please take a look

File: tests/namespace/test_mibs.py (hunk 2 above)
Comment (ysmanman) [reply tests/namespace/test_mibs.py:40]: @SuvarnaMeenakshi extended test to verify the oids of recirc ports.

Comment (ysmanman) [reply tests/namespace/test_mibs.py:40]: @SuvarnaMeenakshi I extended test to verify the oids of recirc ports.

**Comment by ysmanman** [comment]
> @ysmanman The test `test_init_sync_d_interface_tables_for_recirc_ports` is failing. Please take a look

@arlakshm I fixed the test and it passed now.

**Review by arlakshm — APPROVED** [review]

File: tests/namespace/test_mibs.py (hunk 2 above)
Comment (SuvarnaMeenakshi) [reply tests/namespace/test_mibs.py:40]: https://github.com/Azure/sonic-snmpagent/issues/256 - created this issue to track this.

**Review by SuvarnaMeenakshi — APPROVED** [review]
lgtm

**Pull request merged**